opensourceleg/actuators/dephy.py

```diff
@@ -0,0 +1,1739 @@
+        """
+        return self._thermal_scale
+
+
+def _dephy_legacy_voltage_mode_entry(dephy_actuator: "DephyActuator") -> None:
+    LOGGER.debug(msg=f"[DephyControlMode] Entering Voltage control mode.")
+
+
+def _dephy_legacy_voltage_mode_exit(dephy_actuator: "DephyActuator") -> None:
+    LOGGER.debug(msg=f"[DephyControlMode] Exiting Voltage control mode.")
+    dephy_actuator.set_motor_voltage(value=0)
+    time.sleep(DEPHY_SLEEP_DURATION)
+
+
+def _dephy_legacy_current_mode_entry(dephy_actuator: "DephyActuator") -> None:
+    LOGGER.debug(msg=f"[DephyControlMode] Entering Current control mode.")
+
+
+def _dephy_legacy_current_mode_exit(dephy_actuator: "DephyActuator") -> None:
+    LOGGER.debug(msg=f"[DephyControlMode] Exiting Current control mode.")
+    dephy_actuator.set_control_mode(mode=CONTROL_MODES.VOLTAGE)
+    dephy_actuator.set_motor_voltage(value=0)
+    time.sleep(DEPHY_SLEEP_DURATION)
+
+
+def _dephy_legacy_position_mode_entry(dephy_actuator: "DephyActuator") -> None:
+    LOGGER.debug(msg=f"[DephyControlMode] Entering Position control mode.")
+
+
+def _dephy_legacy_position_mode_exit(dephy_actuator: "DephyActuator") -> None:
+    LOGGER.debug(msg=f"[DephyControlMode] Exiting Position control mode.")
+    dephy_actuator.set_control_mode(mode=CONTROL_MODES.VOLTAGE)
+    dephy_actuator.set_motor_voltage(value=0)
+    time.sleep(DEPHY_SLEEP_DURATION)
+
+
+def _dephy_legacy_impedance_mode_entry(dephy_actuator: "DephyActuator") -> None:
+    LOGGER.debug(msg=f"[DephyControlMode] Entering Impedance control mode.")
+
+
+def _dephy_legacy_impedance_mode_exit(dephy_actuator: "DephyActuator") -> None:
+    LOGGER.debug(msg=f"[DephyControlMode] Exiting Impedance control mode.")
+    dephy_actuator.set_control_mode(mode=CONTROL_MODES.VOLTAGE)
+    dephy_actuator.set_motor_voltage(value=0)
+    time.sleep(DEPHY_SLEEP_DURATION)
+
+
+DEPHY_LEGACY_CONTROL_MODE_CONFIGS = CONTROL_MODE_CONFIGS(
+    POSITION=ControlModeConfig(
+        entry_callback=_dephy_legacy_position_mode_entry,
+        exit_callback=_dephy_legacy_position_mode_exit,
+        has_gains=False,
+        max_gains=ControlGains(kp=1000, ki=1000, kd=1000, k=0, b=0, ff=0),
+    ),
+    CURRENT=ControlModeConfig(
+        entry_callback=_dephy_legacy_current_mode_entry,
+        exit_callback=_dephy_legacy_current_mode_exit,
+        has_gains=False,
+        max_gains=ControlGains(kp=80, ki=800, kd=0, k=0, b=0, ff=128),
+    ),
+    VOLTAGE=ControlModeConfig(
+        entry_callback=_dephy_legacy_voltage_mode_entry,
+        exit_callback=_dephy_legacy_voltage_mode_exit,
+        has_gains=False,
+        max_gains=None,
+    ),
+    IMPEDANCE=ControlModeConfig(
+        entry_callback=_dephy_legacy_impedance_mode_entry,
+        exit_callback=_dephy_legacy_impedance_mode_exit,
+        has_gains=False,
+        max_gains=ControlGains(kp=80, ki=800, kd=0, k=1000, b=1000, ff=128),
+    ),
+)
+
+
+class DephyLegacyActuator(ActuatorBase, Device):
```

Comment:
What is the legacy actuator vs the normal one?

Reply:
The DephyLegacyActuator corresponds to flexsea <=8.0.1

Moving forward especially with RPi5 bringing python3.11 as default python, I recommend DephyActuator class (corresponds to flexsea 12.0.3, which supports 1000Hz and includes major bug fixes within their compiled code) should be the new default.

Reply:
@jderosia has an incoming PR that refactors this class to inherit from the DephyActuator class and patch the methods that are different to reduce the amount of duplicated code.